web-console/knikubevirt/vm-templates/vm-templates.md

```diff
@@ -0,0 +1,89 @@
+
+### Summary
+
+- "Virtual Machine Templates" are located under "Virtual Machines" in the primary navigation.
+- VM-templates displays a list view similar to the VM list view with relevant changes in columns and actions.
+- From this list the user can create a new template, create a virtual machine from a template, edit a template, or delete a template.
+- Clicking on a template will bring the user to the template’s Overview tab (TBD).
+Creating or editing a template will display a wizard similar to the “No Template state” wizard from Create VM but with slight adjustments.
+- Users can also create templates based on existing VMs from the actions menu of any VM within the VM List View.
+
+
+
+## Template list view
+
+![Offline Template List View](img/1-0-vm-template-list.jpg)
+
+The user will be able to view a list of all templates that exist. From this list the user can create a new template, create a virtual machine from a template, edit a template, or delete a template.
+
+
+
+
+## Create template from primary button
+
+
+![Offline Template List View](img/1-1-0-create-template.jpg)
+
+Clicking “Create Template” will display a wizard similar to the “No Template state” wizard from Create VM but with slight adjustments.
+
+![Offline Template List View](img/1-1-1-create-template-modal.jpg)
+
+1. Modal title changes to “Create Template”
+2. Choosing a template from a dropdown is not available
+3. The two checkboxes for “Start virtual machine on creation” and “Create new template from configuration” are removed
+4. The quick action button at the bottom is now “Create Template”
+5. In Step 5 “Review”, references to “Virtual Machine” are replaced with “Template”
+
+    a. Create/Creating Virtual Machine -> Create/Creating Template
+
+    b. Virtual Machine creation failed -> Template creation failed
+
+    c. Virtual machine created successfully -> Template created successfully
+
+Everything else remains the same.
+
+
+## Create template from VM List
+
+
+![Offline Template List View](img/1-2-0-create-template-from-vm-list.jpg)
+
+Users can also create templates based on existing VMs from the actions menu of any VM within the VM List View. 
+
+![Offline Template List View](img/1-2-1-create-template-from-vm-list-modal.jpg)
+
+The VM they choose pre-fills the Provision Source, OS, Flavor, Workload Profile, and all other Networking, Storage, and Advanced settings of the Create Template wizard.
+
+## Create VM from another template
+
+
+![Offline Template List View](img/1-3-0-create-template-from-template-list.jpg)
+
+Users can initiate the Create VM wizard from the actions menu of a VM template.
+
+![Offline Template List View](img/1-3-1-create-template-from-template-list-modal.jpg)
+
+The template they choose is preselected and the Provision Source, OS, Flavor, and Workload Profiles are filled in. Refer to the Create VM Wizard for all other steps.
+
+
+
+
+![Offline Template List View](img/1-2-2-create-template-from-vm-list-starting.jpg)
```

Comment:
Same comment as pervious PRs. Not sure we need to tell users this has started if the action was just taken by the user and the status changes. Seems like enough feedback.

Reply:
@beanh66 
I agree that this behavior is not ideal, especially when we land the user on a VM list:

- Using a modal, we don't want to move the user to another page on 'Applying' that modal, but to return to the page in the background. Plus, there is no benefit of landing on the VM list. If it was the new VM, it would be a different case.
- When 'Applying' the modal, the user does need feedback to know that something happened.  But is enough to display the toast notification as a 'success' feedback, and add a link to 'View new VM' page. There is no need to land her out of context.

WDYT?

@matthewcarleton this 'landing on a different page from applying modal' also appears in other places, like VM snapshots. We need to align whatever we decide here to those other places as a rule of thumb

Reply:
I agree.
I think that sending the user to another page after 'applying' a modal is an issue altogether. The user should return to the background page.
On returning to the list view page, the user can view the new item on the list.
It is not solid feedback since we don't highlight that new item, but it is enough to remove the toast.

Reply:
I agree on removing the toast.

Sending the user to another page after 'applying' a modal is an issue. The user should return to the background page. On returning to the list view page, the user can view the new item on the list.
It is not solid feedback since we don't highlight that new item, but it is enough to remove the toast.

@matthewcarleton this 'landing on a different page from applying modal' also appears in other places, like VM snapshots. We need to align whatever we decide here to those other places as a rule of thumb